google-http-client-apache-v5/src/main/java/com/google/api/client/http/apache/v5/Apache5ResponseContent.java

```diff
@@ -0,0 +1,36 @@
+package com.google.api.client.http.apache.v5;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+
+public class Apache5ResponseContent extends InputStream {
+  private final ClassicHttpResponse response;
+  private final InputStream wrappedStream;
+
+  public Apache5ResponseContent(InputStream wrappedStream, ClassicHttpResponse response) {
```

Comment:
should we write a javadoc to explain the function of this class?

Reply:
yes, indeed. I added a quick javadoc

Reply:
sorry to nit, but I want to make sure the next developers understand why this decision was made, how about?  "The main purpose is to be able to close the response as well as the content input stream when {@link #close()} is called, in order to not break the existing contract with clients using apache v4 that only required them to close the input stream to clean up all resources."

Reply:
Makes sense to make it more clear why are we doing this, thanks! Updated.